zcash_primitives/src/transaction/components/sapling.rs

```diff
@@ -40,49 +117,84 @@ impl std::fmt::Debug for SpendDescription {
     }
 }
 
-impl SpendDescription {
-    pub fn read<R: Read>(mut reader: &mut R) -> io::Result<Self> {
-        // Consensus rules (§4.4):
-        // - Canonical encoding is enforced here.
-        // - "Not small order" is enforced in SaplingVerificationContext::check_spend()
-        //   (located in zcash_proofs::sapling::verifier).
-        let cv = {
-            let mut bytes = [0u8; 32];
-            reader.read_exact(&mut bytes)?;
-            let cv = jubjub::ExtendedPoint::from_bytes(&bytes);
-            if cv.is_none().into() {
-                return Err(io::Error::new(io::ErrorKind::InvalidInput, "invalid cv"));
-            }
-            cv.unwrap()
-        };
+/// Consensus rules (§4.4) & (§4.5):
+/// - Canonical encoding is enforced here.
+/// - "Not small order" is enforced in SaplingVerificationContext::(check_spend()/check_output())
+///   (located in zcash_proofs::sapling::verifier).
+pub fn read_point<R: Read>(mut reader: R, field: &str) -> io::Result<jubjub::ExtendedPoint> {
+    let mut bytes = [0u8; 32];
+    reader.read_exact(&mut bytes)?;
+    let point = jubjub::ExtendedPoint::from_bytes(&bytes);
+
+    if point.is_none().into() {
+        Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "invalid ".to_owned() + field,
```

Comment:
```suggestion
            format!("invalid {}", field),
```
This will return a `String`.